### ISSUE TITLE
Synchronised input groups

### ISSUE THREAD

**Issue by alexwhitman** [gnunn1/terminix#279]
Synchronised input is currently done at the session level but it would be useful if terminals could be grouped within a single session and input synchronisation done at the group level.

**Comment by gnunn1** [comment]
@alexwhitman I added the feature some time ago that allows you to decide what terminals in a session participate in synchronization. When the titlebar is visible and session input synchronization is on, there is a little keyboard icon that appears that you can toggle on/off to include/exclude specific terminals. This feature is also available in the terminal context menu when the titlebar is off.

I wasn't planning on taking this any further, as I think allowing users to start creating and managing arbitrary groups would be confusing and not intuitive. Plus I'd need to come up with a way to show the different groups visually and I don't have a good idea for that.

Thoughts?

**Comment by alexwhitman** [comment]
It's a feature I've used quite a bit in terminator as I tend to have it maximised on a single large monitor which then allows for comfortably fitting quite a few terminals.

The screenshot below shows how it's done in terminator to distinguish/manage each group but I can understand why it might be something you wouldn't want to add.

![selection_001](https://cloud.githubusercontent.com/assets/422013/14832422/2a576b6e-0bf2-11e6-8b3a-dc43cfb0d102.png)

**Comment by gnunn1** [comment]
Thanks, I still tend to think it's confusing and not something to many folks would use, but I'll leave this open as a means to judge interest.

**Comment by mattwilmott** [comment]
+1 for this feature as well. I use this at work probably 2 or 3 times a day when I don't feel like breaking out ansible. Easily SSH to a handful of servers and broadcast

**Comment by dansouza** [comment]
One thing I miss is the ability to broadcast keystrokes across separate instances/windows of Terminix, like it is possible with Terminator. For example, imagine you have a two 1920x1080 monitors and on one monitor you have a Terminix window with 4 terminals (2x2), then on another monitor another Terminix window the same way, having keystrokes broadcasted to all terminals across the windows based either on groups (like Terminator does).

Having 8 terminals (4x2) on a single 1920x1080 monitor is clumsy and 4K monitors are still expensive if you factor in the new GPU. =)

**Comment by zatan** [comment]
+1

**Comment by cazacugmihai** [comment]
+1

**Comment by KmBKeef** [comment]
+1

**Comment by mistersister** [comment]
+1 

Also a little offtopic but when you sync terminals on a session and decide to leave one out by clicking on the icon in the title bar, I find it impossible to tell if its excluded. The icon should be way more dimmed or totally change colors to something like red for instance.

![screenshot_2017-08-03_17-00-20](https://user-images.githubusercontent.com/3819621/28928629-fa432f96-786d-11e7-936e-6bd054a08738.png)

First one in red would be an ideal way to easily tell if that terminal is not in sync. The second icon is an excluded terminal. If you compare it with the third icon you can see that is difficult to tell if its grouped or not.

**Comment by gnunn1** [comment]
@mistersister I think that's a problem with your theme, try it with Adwaita so see how it looks there.

**Comment by mistersister** [comment]
@gnunn1 yeah, it seems like my current theme (Numix) places no borders around the icon unlike Adwaita. Damn you material design! I tested some more random themes installed and some had the same issue, others didnt. 

Maybe you should consider make it so that is less theme dependent? It sucks when theming gets in the way of usability.

**Comment by dfbadawi** [comment]
+1

**Comment by krishraghuram** [comment]
I agree with @dansouza 

It would be great if we could broadcast across sessions and windows.

At my work, I have a small monitor, so I stick to two terminals stacked vertically. 
To compensate, I open 3-4 sessions.

**Comment by nubunto** [comment]
+1 for this feature. Maybe one could have this functionality without having to wire it in Tilix itself. Register new sessions over a socket, socket executes commands it receives, maybe?

**Comment by chrisbrake** [comment]
+1

**Comment by akaHuman** [comment]
+1. I think this is a pretty useful feature to have. I migrated from Terminator to Tilix. This is one feature I find missing. It comes in pretty handy if you need to ssh to multiple hosts and run the same commands on all of them.

**Comment by donfuz14** [comment]
+1 for this feature if I'm allowed to ;)

**Comment by EtreRe** [comment]
+1

**Comment by hboetes** [comment]
This is an essential feature for which I still have to use terminator.

**Comment by fernandolobo** [comment]
+1

**Comment by rafaelchavesfreitas** [comment]
+1

**Comment by seocam** [comment]
Just a note for the +1: instead of adding "+1" comments use reactions in issue description. This way the maintainers will be able sort by reactions.

On the other hand, adding comments explaining why this feature would be helpful for you could help maintainers to understand the actual needs and plan accordingly to that.

**Comment by seocam** [comment]
I used to use this feature in terminator and iterm2 to witch a group of terminals to a different directory and python virtualenv. That kind of context change happens many times in a single day and that's where I miss the feature the most.

Broadcasting is also useful for me if I need to run a command in a few different VMs. This second scenario it's really used when I only have a few vms (4 at most), if it's more than that I'd use Ansible or Puppet...

**Comment by Zeioth** [comment]
+1, I find it tremendously powerful in some scenarios.

**Comment by zeandrade** [comment]
+ 1

**Comment by euikook** [comment]
+1

**Comment by marschal** [comment]
+1

**Comment by mhalano** [comment]
Any news about this feature? I just turned to Terminator because it has support for broadcast where I can type the same command on multiple panes.

**Comment by zeandrade** [comment]
I'm using V 1.9.1 in Ubuntu , and I have a "Synchronize Entries" Option on menu. When enabled, it's create a keyboard icon, that's enable/disable the option for all splitted terminals present.

**Comment by valderaplets** [comment]
some news ? it's really convenient when, with 6 terminals for example, you want to make 3 groups of two (two masters, two satellites and two agent for example)

**Comment by ricardobranco777** [comment]
> I'm using V 1.9.1 in Ubuntu , and I have a "Synchronize Entries" Option on menu. When enabled, it's create a keyboard icon, that's enable/disable the option for all splitted terminals present. ( tilix 1.9.3-4build3 , tilix-common 1.9.3-4build3)

This!
